toolshed/routes/ai/llm/generateText.ts

```diff
@@ -12,13 +22,105 @@ export interface GenerateTextParams {
   stop_token?: string;
   abortSignal?: AbortSignal;
   max_tokens?: number;
+  jsonMode?: boolean;
 }
 
 export interface GenerateTextResult {
   message: { role: "user" | "assistant"; content: string };
   stream?: ReadableStream;
 }
 
+// Configure the model parameters for JSON mode based on provider
+function configureJsonMode(
+  streamParams: Record<string, any>,
+  modelName: string,
+  messages: { role: "user" | "assistant"; content: string }[],
+  isStreaming: boolean,
+): void {
```

Comment:
Thoughts on adding tests for this? It's got a bunch of specific edge cases so tests would act as documentation of expected behavior etc.

Might be generateable?